socs/agents/galildmc_axis_controller/agent.py

```diff
@@ -0,0 +1,1170 @@
+
+        return True, f'Gearing ratio for axis {axis} is {resp}'
+
+    @ocs_agent.param('configfile', type=str, default=None)
+    def input_configfile(self, session, params=None):
+        """input_configfile(configfile=None)
+
+        **Task** Upload GalilDMC Axis Controller configuration file to initialize device
+        and axes on device
+
+        Parameters:
+            configfile (str, optional):
+                name of .yaml config file. Defaults to the file set in the site config.
+
+
+        """
+        configfile = params['configfile']
+        if configfile is None:
+            configfile = self.configfile
+        if configfile is None:
+            raise ValueError("No configfile specified")
+        configfile = os.path.join(os.environ['OCS_CONFIG_DIR'], configfile)
+
+        with self.lock.acquire_timeout(timeout=5, job='input_configfile') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            cfg = read_config(configfile)
+            try:
+                gal = cfg.get('galil', {})
+                self.cfg = cfg
+                self.motorsettings = gal.get('motorsettings', {})
+                self.axis_map = gal.get('motorconfigparams', {})
+                self.axes = list(self.axis_map.keys())
+                self.brakes = gal['brakes']['output_map']
+            except Exception as e:
+                return False, f'Config parse error: {e}'
+
+            for a in self.axes:
+                # set motor type
+                motortype = self.axis_map[a]['MT']
+                self.stage.set_motor_type(axis=a, motortype=motortype)
+
+                # set off on error
+                errtype = self.axis_map[a]['OE']
+                errtype = int(errtype)
+                self.stage.set_off_on_error(axis=a, errtype=errtype)
+
+                # set amp gain
+                gn = self.axis_map[a]['AG']
+                self.stage.set_amp_gain(axis=a, val=gn)
+
+                # set torque limit
+                tl = self.axis_map[a]['TL']
+                self.stage.set_torque_limit(axis=a, val=tl)
+
+                # set current loop gain
+                clgn = self.axis_map[a]['AU']
+                self.stage.set_amp_currentloop_gain(axis=a, val=clgn)
+
+                # enable sin commutation
+                initstate = gal['initaxisparams']['BA']
+                if initstate == 'True':
+                    # enable sin commutation
+                    self.stage.enable_sin_commutation(axis=a)
+                    # set magnetic cycle
+                    mag = gal['initaxisparams']['BM']
+                    self.stage.set_magnetic_cycle(axis=a, val=mag)
+                    # set dwell times before initializing
+                    self.stage.set_dwell_times(t_first=self.first_dwell, t_second=self.sec_dwell)
```

Comment:
`self.first_dwell` and `self.sec_dwell` here are defined in `init`. If this function were to be run as a standalone at a later time, and the config file itself has changed, then these changed values would still be the old values, not the new ones.

Reply:
oops, thanks for pointing out the typos!